### PR TITLE
feat: added bo integration case references ff

### PR DESCRIPTION
**Pull request by Yakitrak**
Ticket: https://pins-ds.atlassian.net/browse/ASB-2244

PR adds consolidated feature flag for all back office integrations case references: `BACK_OFFICE_API_INTEGRATION_CASE_REFERENCES`

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] My code changes do not include any hardcoded secrets or passwords
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My Tech Lead is aware of any infrastructure changes that will cost money

### CODE REVIEW

File: app/stacks/uk-south/applications-service/README.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ This component contains the infrastructure required for the applications service
 | <a name="input_app_service_plan_id"></a> [app\_service\_plan\_id](#input\_app\_service\_plan\_id) | The id of the app service plan | `string` | n/a | yes |
 | <a name="input_applications_service_public_url"></a> [applications\_service\_public\_url](#input\_applications\_service\_public\_url) | The public URL for the Applications Service frontend web app | `string` | n/a | yes |
 | <a name="input_applications_service_vpn_gateway_shared_key"></a> [applications\_service\_vpn\_gateway\_shared\_key](#input\_applications\_service\_vpn\_gateway\_shared\_key) | The applications service virtual network gateway shared key | `string` | n/a | yes |
+| <a name="input_back_office_integration_case_references"></a> [back\_office\_integration\_case\_references](#input\_back\_office\_integration\_case\_references) | list of case references (comma separated) that use back office for submissions | `string` | n/a | yes |
```
Comment (Philip-Philippou1) [inline app/stacks/uk-south/applications-service/README.md:67]: list of case references (comma separated) that use back office 
I would remove "for submissions"

File: app/stacks/uk-south/applications-service/variables.tf [hunk 2]
```diff
@@ -70,6 +70,11 @@ variable "back_office_integration_get_applications" {
   type        = string
 }
 
+variable "back_office_integration_case_references" {
+  description = "list of case references (comma separated) that use back office for submissions"
```
Comment (Philip-Philippou1) [inline app/stacks/uk-south/applications-service/variables.tf:74]: I would remove the "for submissions"

File: app/stacks/uk-west/applications-service/README.md [hunk 3]
```diff
@@ -64,6 +64,7 @@ This component contains the infrastructure required for the applications service
 | <a name="input_app_service_plan_id"></a> [app\_service\_plan\_id](#input\_app\_service\_plan\_id) | The id of the app service plan | `string` | n/a | yes |
 | <a name="input_applications_service_public_url"></a> [applications\_service\_public\_url](#input\_applications\_service\_public\_url) | The public URL for the Applications Service frontend web app | `string` | n/a | yes |
 | <a name="input_applications_service_vpn_gateway_shared_key"></a> [applications\_service\_vpn\_gateway\_shared\_key](#input\_applications\_service\_vpn\_gateway\_shared\_key) | The applications service virtual network gateway shared key | `string` | n/a | yes |
+| <a name="input_back_office_integration_case_references"></a> [back\_office\_integration\_case\_references](#input\_back\_office\_integration\_case\_references) | list of case references (comma separated) that use back office for submissions | `string` | n/a | yes |
```
Comment (Philip-Philippou1) [inline app/stacks/uk-west/applications-service/README.md:67]: same here

File: app/stacks/uk-west/applications-service/variables.tf [hunk 4]
```diff
@@ -140,6 +140,11 @@ variable "back_office_integration_get_applications" {
   type        = string
 }
 
+variable "back_office_integration_case_references" {
+  description = "list of case references (comma separated) that use back office for submissions"
```
Comment (Philip-Philippou1) [inline app/stacks/uk-west/applications-service/variables.tf:144]: Same here.

**Comment by Yakitrak** [comment]
Thanks for the spot @Philip-Philippou1

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**